### ISSUE TITLE
Ensemble API: Integrate probabilistic forecasts

### ISSUE THREAD

**Issue by patrick-zippenfenig** [open-meteo/open-meteo#349]
As suggested by @phillipseamore the Ensemble API (see #81) can be extended by probabilistic weather forecasts to calculate the likelihood of a given weather condition directly in the API.

Something like:
`p~[parameter]~[comparison]~[threshold]`
With an upper and lower comparison:
`p~[parameter]~[comparison1]~[threshold1]~[comparison2]~[threshold2]`

Comparison could be:
`morethan` = ">"
`lessthan` = "<"
`moreeq` = ">="
`lesseq` = "<="
`eq` = "=="

Working with the limitations we have of available safe symbols to use as seperators I'd suggest a tilde "~" since we have the underscore already in parameters and need hyphen and dot for values (floats and negative).
Having a "p" (for probabilities) at the beginning would make this same scheme available for something else in the future since we won't have any more safe symbols to use.

To get the probability of windspeed_10m above 25 m/s would be: `p~windspeed_10m~more~25`
To get the probability of temperature below -10 would be: `p~temperature_2m~less~-10`
To get the probability of precipitation being more than or equal to 0.3: `p~precipitation~moreeq~0.3`
To get the probability of precipitation being less than or equal to 0.1: `p~precipitation~lesseq~0.1`
To get the probability of precipitation being none: `p~precipitation~eq~0`

Double combinations (eq not allowed):
To get the probability of total cloud cover >=50 && <80: `p~cloudcover~moreeq~50~less~80`

If a request has a parameter than includes "~" we split it.
It is invalid when:
- first match isn't "p"
- it isn't exactly 4 or 6 parts
- the parameter doesn't exist
- comparison isn't in list
- has 6 parts and the first 4 characters of both comparisons are the same ("more"/"less")
- has 6 parts and one comparison is "eq"

```
split = "p~cloudcover~moreeq~50~less~80".split("~")
split[0] = "p" for probabilities
split[1] = parameter to lookup
split[2] = 1st comparison operator
split[3] = 1st value to compare
split[4] = 2nd comparison operator
split[5] = 2nd value to compare
```

Here's a quick idea for expanding the scheme.

e = extrema/max
m = minima
d = standard deviation/spread
q = quantile
a = average/mean
n = median
o = mode

Example for a user doing a request to use in a box & whiskers plot of temperature would be:
```
e~temperature_2m     -- whisker max
m~temperature_2m     -- whisker min
q~temperature_2m~10  -- outer box min
q~temperature_2m~90  -- outer box max
q~temperature_2m~25  -- inner box min
q~temperature_2m~75  -- inner box max
n~temperature_2m~75  -- median
```

Now we would need to extend the validation:
- has 2 or 3 parts
- first match is p, e, m, d, q, a, n or o
- if other than q and not exactly 2 parts, invalid
- if q isn't exactly 3 parts, invalid
- if q, threshold should be >=0 && <= 100

**Comment by lutzvdb** [comment]
A sidenote: Such aggregations would also be interesting for other APIs. Currently, I have to download the entire (daily) history in order to provide visualizations for aggregates (see climatevisualizer). If I had the possibility to request monthly aggregates (min/max for example), that could save on bandwidth. It could be interesting to have a look at the ODATA standard for generalized querying - however, I am not sure if it will integrate well with existing endpoint definitions.

**Comment by patrick-zippenfenig** [comment]
@lutzvdb monthly aggregations could be quite useful and save a lot of bandwidth. Maybe it even makes sense to store monthly data on the server.

Do you have an example how to use ODATA for weather data?

**Comment by lutzvdb** [comment]
No, I don't have a concrete example for that unfortunately. It just came to my mind when I saw your notes about devising a scheme how to query weather variables.
